File: Ruby/response/sms/InboundSMSMessage.rb

class InboundSMSMessage

	def initialize
		@dateTime=nil
		@destinationAddress=nil
		@senderAddress=nil
		@messageId=nil
		@message=nil
		@resourceURL=nil
	end

	def initializeJSON(jsondict)
		@dateTime=nil
		if (jsondict!=nil) && (jsondict.has_key?'dateTime') && (jsondict['dateTime']!=nil)
			@dateTime=jsondict['dateTime']
		end
		@destinationAddress=nil
		if (jsondict!=nil) && (jsondict.has_key?'destinationAddress') && (jsondict['destinationAddress']!=nil)
			@destinationAddress=jsondict['destinationAddress']
		end
		@senderAddress=nil
		if (jsondict!=nil) && (jsondict.has_key?'senderAddress') && (jsondict['senderAddress']!=nil)
			@senderAddress=jsondict['senderAddress']
		end
		@messageId=nil
		if (jsondict!=nil) && (jsondict.has_key?'messageId') && (jsondict['messageId']!=nil)
			@messageId=jsondict['messageId']
		end
		@message=nil
		if (jsondict!=nil) && (jsondict.has_key?'message') && (jsondict['message']!=nil)
			@message=jsondict['message']
		end
		@resourceURL=nil
		if (jsondict!=nil) && (jsondict.has_key?'resourceURL') && (jsondict['resourceURL']!=nil)
			@resourceURL=jsondict['resourceURL']
		end
	end

	def getDateTime
		@dateTime
	end

	def setDateTime(dateTime)
		@dateTime=dateTime
	end

	def getDestinationAddress
		@destinationAddress
	end

	def setDestinationAddress(destinationAddress)
		@destinationAddress=destinationAddress
	end

	def getSenderAddress
		@senderAddress
	end

	def setSenderAddress(senderAddress)
		@senderAddress=senderAddress
	end

	def getMessageId
		@messageId
	end

	def setMessageId(messageId)
		@messageId=messageId
	end

	def getMessage
		@message
	end

	def setMessage(message)
		@message=message
	end

	def getResourceURL
		@resourceURL
	end

	def setResourceURL(resourceURL)
		@resourceURL=resourceURL
	end


end 

File: Python/response/sms/InboundSMSMessage.py

class InboundSMSMessage:
	"""contains the message details for an SMS message"""
	
	def __init__(self):
		"""Default class constructor"""
		self.dateTime=None
		self.destinationAddress=None
		self.senderAddress=None
		self.messageId=None
		self.message=None
		self.resourceURL=None
	
	def __init__(self, jsondict):
		"""Class constructor that will create an instance initialised from a parsed JSON data block"""
		self.dateTime=None
		if jsondict is not None and 'dateTime' in jsondict and jsondict['dateTime'] is not None:
			self.dateTime=jsondict['dateTime']
		self.destinationAddress=None
		if jsondict is not None and 'destinationAddress' in jsondict and jsondict['destinationAddress'] is not None:
			self.destinationAddress=jsondict['destinationAddress']
		self.senderAddress=None
		if jsondict is not None and 'senderAddress' in jsondict and jsondict['senderAddress'] is not None:
			self.senderAddress=jsondict['senderAddress']
		self.messageId=None
		if jsondict is not None and 'messageId' in jsondict and jsondict['messageId'] is not None:
			self.messageId=jsondict['messageId']
		self.message=None
		if jsondict is not None and 'message' in jsondict and jsondict['message'] is not None:
			self.message=jsondict['message']
		self.resourceURL=None
		if jsondict is not None and 'resourceURL' in jsondict and jsondict['resourceURL'] is not None:
			self.resourceURL=jsondict['resourceURL']
	
	def getDateTime(self):
		"""getter for dateTime : date and time that the message was sent"""
		return self.dateTime
	
	def setDateTime(self,dateTime):
		"""setter for dateTime : date and time that the message was sent"""
		self.dateTime=dateTime
	
	def getDestinationAddress(self):
		"""getter for destinationAddress : MSISDN or other address of the message recipient"""
		return self.destinationAddress
	
	def setDestinationAddress(self,destinationAddress):
		"""setter for destinationAddress : MSISDN or other address of the message recipient"""
		self.destinationAddress=destinationAddress
	
	def getSenderAddress(self):
		"""getter for senderAddress : MSISDN or other address of the message sender"""
		return self.senderAddress
	
	def setSenderAddress(self,senderAddress):
		"""setter for senderAddress : MSISDN or other address of the message sender"""
		self.senderAddress=senderAddress
	
	def getMessageId(self):
		"""getter for messageId : contains the server message Id field"""
		return self.messageId
	
	def setMessageId(self,messageId):
		"""setter for messageId : contains the server message Id field"""
		self.messageId=messageId
	
	def getMessage(self):
		"""getter for message : the message body"""
		return self.message
	
	def setMessage(self,message):
		"""setter for message : the message body"""
		self.message=message
	
	def getResourceURL(self):
		"""getter for resourceURL : specifies the URL to the message on the server"""
		return self.resourceURL
	
	def setResourceURL(self,resourceURL):
		"""setter for resourceURL : specifies the URL to the message on the server"""
		self.resourceURL=resourceURL
	
